src/components/k8s/pcn_k8s/controllers/event_dispatcher.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"sync"
+
+	pcn_types "github.com/SunSince90/polycube/src/components/k8s/pcn_k8s/types"
+)
+
+// EventDispatcher dispatches the event to all subscribers
+type EventDispatcher struct {
+	name        string
+	n           pcn_types.SubscriberID
+	subscribers map[pcn_types.SubscriberID]subscriber
+	lock        sync.Mutex
+}
+
+// EventDispatchersContainer contains the three basic EventDispatcher-s: New, Update, Delete
+type EventDispatchersContainer struct {
+	new    *EventDispatcher
+	update *EventDispatcher
+	delete *EventDispatcher
+}
+
+type subscriber func(interface{})
+
+// NewEventDispatcher starts a new event dispatcher
+func NewEventDispatcher(name string) *EventDispatcher {
+	return &EventDispatcher{
+		name:        name,
+		n:           0,
+		subscribers: make(map[pcn_types.SubscriberID]subscriber),
+	}
+}
+
+// Dispatch will dispatch the event to the list of subscribers
+func (d *EventDispatcher) Dispatch(item interface{}) {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+
+	//	Are there any subscribers?
+	if (len(d.subscribers)) < 1 {
+		return
+	}
+
+	//	Loop through all of the subscribers
+	for _, s := range d.subscribers {
+		// The controller will make it go
+		//	go s(item)
+		s(item)
+	}
+}
+
+// Add will add a new subscriber
+func (d *EventDispatcher) Add(s subscriber) pcn_types.SubscriberID {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+
+	d.subscribers[d.n+1] = s
```

Comment:
I used this approach to keep it a bit simpler, as the ID is never decremented. So I should always have the guarantee that that number never exists.
I didn't look that much into random generated IDs, I just used this because it is simpler... I also think that maybe it could cause some delays in its generation (but I'm sure I'm overthinking this, this delay probably is not that relevant)

Reply:
That number could overflow and start replacing existing entries, not sure however how much time it'll be needed for this to overflow. 
The delays in the generation of a random number should be not a big deal here.